ethereumj-core/src/main/java/org/ethereum/sharding/crypto/BLS381.java

```diff
@@ -0,0 +1,53 @@
+package org.ethereum.sharding.crypto;
+
+import java.math.BigInteger;
+
+/**
+ * Common interface for BLS381-12 implementation
+ */
+public interface BLS381 {
+
+    BI generatePrivate();
+
+    BI restorePrivate(BigInteger value);
+
+    BI restorePrivate(byte[] value);
+
+    ECP1Point restoreECP1(byte[] value);
+
+    ECP2Point restoreECP2(byte[] value);
+
+    ECP2Point generator2();
+
+    ECP1Point mapToECP1(byte[] value);
+
+    FP12Point pair(ECP2Point pointECP2, ECP1Point pointECP1);
+
+    interface BI {
+        BigInteger asBigInteger();
```

Comment:
Should be removed in case if we're not sticking with any specific big number implementations